scripts/install

```diff
@@ -103,8 +103,8 @@ ynh_script_progression --message="Installing yarn dependency and building the ap
 pushd $install_dir
 	ynh_use_nodejs
 	ynh_exec_warn_less ynh_exec_as $app $ynh_node_load_PATH yarn install
-	ynh_exec_warn_less ynh_exec_as $app $ynh_node_load_PATH yarn add nuxt
-	ynh_exec_warn_less ynh_exec_as $app $ynh_node_load_PATH $ynh_npm run build --config $install_dir/config.json #There is no "install" command, only build
+	#ynh_exec_warn_less ynh_exec_as $app $ynh_node_load_PATH yarn add nuxt
```

Comment:
Do you know how to fix this ?

Reply:
there is no need to build gancio, the package is kind of pre built you only need to install it as described [in documentation](https://gancio.org/install/debian).
`yarn global add --network-timeout 1000000000 --silent https://gancio.org/latest.tgz`

where `latest.tgz` is the archive yunohost downloaded via releases.source configuration so this could probably be used instead. `yarn install` is to install dependencies as you will need it anyway.

Then you should specify a way to start it.. I've seen in install script that there is a systemd .service involved? this could be taken from [here](http://gancio.org/gancio.service).

hope it helps

Reply:
> you only need to install it as described in documentation.
yarn global add --network-timeout 1000000000 --silent https://gancio.org/latest.tgz

We're not going to.
1) no global install, it's unsafe for system stability
2) that's not how installs work in Yunohost.
3) Installing an archive without even a checksum to check for integrity is a security breach.

So that command has to be changed. I don't know how to convert it (I'm not NodeJS-savy).

> yarn install is to install dependencies as you will need it anyway.

Ok, great, thanks for explaining why it's used !

> Then you should specify a way to start it.. I've seen in install script that there is a systemd .service involved?

Already done: https://github.com/YunoHost-Apps/gancio_ynh/blob/testing/conf/systemd.service
It doesn't work.
I could start it manually on my server, but only *after* building it…

Reply:
> We're not going to.
>     1. no global install, it's unsafe for system stability
>     2. that's not how installs work in Yunohost. 
>     3. Installing an archive without even a checksum to check for integrity is a security breach.
> So that command has to be changed. I don't know how to convert it (I'm not NodeJS-savy).

with `yarn` if you use `global` prefix as user it will install the package in user's home, usually `~/.yarn`.
you don't have to do this but it's convenient because yarn also takes care of copying the executable scripts that the package wants to export to a common PATH `~/.yarn/.bin/` (this is declared [here](https://framagit.org/les/gancio/-/blob/master/package.json?ref_type=heads#L121))

so instead of using `yarn global add --network-timeout 1000000000 --silent https://gancio.org/latest.tgz` it is ok to have the archive downloaded as you already doing [here](https://github.com/YunoHost-Apps/gancio_ynh/pull/2/commits/8084be2420c3a579f6b883678b40ed5a15451a1b#diff-06ac558c9cfca9e31a9a235f18d98e6d0c3bfee8bee0088108a29f2c60b318e3R62) , then install dependencies with `yarn install` and then you have to take care of the script's path that will be __INSTALL_DIR__/server/cli.js  (with start as first argument, the cli also supports [other things](https://gancio.org/usage/cli#create-a-new-user).

If you need some other utilities (like changing url) I can easily implement it.

Reply:
>      with yarn if you use global prefix as user it will install the package in user's home, usually ~/.yarn.

Thanks for explaining that. :ok_hand:
In our case, it's the same user for all installs if I remember well (it's root), so that would be an issue. Roughly equals to a global install.

> it is ok to have the archive downloaded as you already doing here , then install dependencies with yarn install and then you have to take care of the script's path that will be INSTALL_DIR/server/cli.js (with start as first argument, the cli also supports other things.

Amazing, thanks a lot !
I'll try that :smiley:

Reply:
Extra question : how do I create admin user with name + password ? I only see a reset password option.

Reply:
if this is mandatory I can add an optional argument to the `gancio users create` command.

Are you copying a pre-filled `config.json`? Because if this is not found it starts with a setup wizard that create an admin user. and also, if there is no user because you skipped the setup, the first one registered automatically became an active administrator.

Reply:
> Are you copying a pre-filled config.json? 

Yes.

> if this is mandatory I can add an optional argument to the gancio users create command.

It seems to be mandatory, as there is no way to create an user for the UI afterwards (I understood it has to be done / activated in the admin UI ?).
That would be awesome !

Reply:
To clarify, we would need to setup the user ADMIN with its ADMIN_PASSWORD argument, as we can prepare that from the install dialog, and generate an user accordingly. Having a fully working admin account would be necessary for us, as far as I understand.

> the first one registered automatically became an active administrator.

I already forgot the exact reason why, but that is not working, no user can be created, using a custom config file.

Reply:
Cool thanks !

Reply:
this is released in [1.6.18](https://gancio.org/changelog#1618---3-nov-23)

Reply:
Thanks a lot !

By the way it's not listed in the releases, but in the tags.